File: biopython_rewrite/blast_fetch_ids.py
#!/usr/bin/env python3
# Toy fetch: from best-hits TSV produce a FASTA with placeholder seqs
import sys
if len(sys.argv)<2:
    print('Usage: blast_fetch_ids.py best_hits.tsv'); sys.exit(1)
with open(sys.argv[1]) as fh:
    for line in fh:
        q,s,*rest = line.strip().split('\t')
        print(f">{s}\nNNNNNNNNNN\n")

File: legacy_perl/blast_fetch_ids.pl
#!/usr/bin/env perl
# Given BLAST best-hits, fetch sequence IDs into a FASTA (toy, requires entrez utils normally)
use strict; use warnings;
my $in = shift or die "Usage: $0 best_hits.tsv\n";
open my $fh,'<',$in or die $!;
while(<$fh>){ chomp; my ($qid,$sid,undef)=split/\t/; print ">${sid}\nNNNNNNNNNN\n"; }
